store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/MiniAccumuloStore.java

```diff
@@ -0,0 +1,274 @@
+            baseDir = new File(FileUtils.getTempDirectory(), BASE_DIRECTORY + UUID.randomUUID());
+        }
+
+        try {
+            FileUtils.deleteDirectory(baseDir);
+            miniAccumuloConfig = new MiniAccumuloConfig(baseDir, ROOTPW);
+            miniAccumuloConfig.setInstanceName(getProperties().getInstance());
+            miniAccumuloCluster = new MiniAccumuloCluster(miniAccumuloConfig);
+            miniAccumuloCluster.start();
+        } catch (final IOException | InterruptedException e) {
+            throw new StoreException(e.getMessage(), e);
+        }
+
+        // Create the user specified in the properties (if not root)
+        // together with the specified password and give them all authorisations
+        try {
+            if (!getProperties().getUser().equalsIgnoreCase("root")) {
+                miniAccumuloCluster.getConnector("root", ROOTPW).securityOperations()
+                        .createLocalUser(getProperties().getUser(), new PasswordToken(getProperties().getPassword()));
+                miniAccumuloCluster.getConnector("root", ROOTPW).securityOperations()
+                        .grantSystemPermission(getProperties().getUser(), SystemPermission.CREATE_TABLE);
+                miniAccumuloCluster.getConnector("root", ROOTPW).securityOperations()
+                        .grantSystemPermission(getProperties().getUser(), SystemPermission.CREATE_NAMESPACE);
+            }
+            Authorizations auths = new Authorizations("public", "private", "publicVisibility", "privateVisibility", "vis1", "vis2");
+            miniAccumuloCluster.getConnector("root", ROOTPW).securityOperations()
+                    .changeUserAuthorizations(getProperties().getUser(), auths);
+        } catch (final AccumuloException | AccumuloSecurityException e) {
+            throw new StoreException(e.getMessage(), e);
+        }
+
+        // Create the new properties object to pass back, including connection items
+        AccumuloProperties accumuloProperties = (AccumuloProperties) properties.clone();
+        accumuloProperties.setInstance(miniAccumuloCluster.getInstanceName());
+        accumuloProperties.setZookeepers(miniAccumuloCluster.getZooKeepers());
+
+        return accumuloProperties;
+    }
+
+    public void tearDownTestDB() {
+        this.closeMiniAccumuloStore();
+    }
+
+    public MiniAccumuloCluster getMiniAccumuloCluster() {
+        return miniAccumuloCluster;
+    }
+
+    public Connector getMiniConnector() {
+        return miniConnector;
+    }
+
+    @Override
+    public void preInitialise(final String graphId, final Schema schema, final StoreProperties properties) throws StoreException {
+        setProperties(properties);
+        // This MiniAccumuloStore object may be being reused so we need to clean up some of the attributes
+        clearAllOperationalHandlers();
+        super.preInitialise(graphId, schema, getProperties());
+    }
+
+    private void clearAllOperationalHandlers() {
```

Comment:
Would this work if operations were added using OperationDeclarations?

Reply:
That situation has tests and they are still passed when the MiniAccumuloStore is used for them.
This was added because tests were failing for operation handlers. It creates a consistent environment for each test in a suite.